nmostesting/mocks/Node.py

```diff
@@ -78,3 +193,261 @@ def node_video_sdp(stream_type):
 
     response.headers["Content-Type"] = "application/sdp"
     return response
+
+@NODE_API.route('/x-nmos', methods=['GET'], strict_slashes=False)
+def x_nmos_root():
+    base_data = ['connection/']
+
+    return make_response(Response(json.dumps(base_data), mimetype='application/json'))
+
+@NODE_API.route('/x-nmos/connection', methods=['GET'], strict_slashes=False)
+def connection_root():
+    base_data = ['v1.0/', 'v1.1/']
+
+    return make_response(Response(json.dumps(base_data), mimetype='application/json'))
+
+@NODE_API.route('/x-nmos/connection/<version>', methods=['GET'], strict_slashes=False)
+def version(version):
+    base_data = ['bulk/', 'single/']
+
+    return make_response(Response(json.dumps(base_data), mimetype='application/json'))
+
+@NODE_API.route('/x-nmos/connection/<version>/single', methods=['GET'], strict_slashes=False)
+def single(version):
+    base_data = ['senders/', 'receivers/']
+
+    return make_response(Response(json.dumps(base_data), mimetype='application/json'))
+
+@NODE_API.route('/x-nmos/connection/<version>/single/<resource>/', methods=["GET"], strict_slashes=False)
+def resources(version, resource):
+    if resource == 'senders':
+        base_data = [r + '/' for r in [*NODE.senders]]
+    elif resource == 'receivers':
+        base_data = [r + '/' for r in [*NODE.receivers]]
+
+    return make_response(Response(json.dumps(base_data), mimetype='application/json'))
+
+@NODE_API.route('/x-nmos/connection/<version>/single/<resource>/<resource_id>', methods=["GET"], strict_slashes=False)
+def connection(version, resource, resource_id):
+    if resource == 'senders':
+        base_data = ["constraints/", "staged/", "active/", "transportfile/", "transporttype/"]
```

Comment:
Technically `transporttype/` should only be returned if the `<version>` is v1.1+, but as I think we're only testing v1.1 in practice at the moment it may just be a note for the future.

Reply:
Right. The IS-05 Controllers test suite ought to have a list of supported spec versions that are user-selectable in the drop-down. The mocks ought to reflect that. If that currently means v1.1 only, that's fine, but the drop-down should only include v1.1 to make that clear. If both Controllers working with v1.0 and v1.1 can be tested, that should be wired up. Same for IS-04, whether it's currently just v1.3 or some/all of the earlier versions.

Given TR-1001-1 currently requires IS-04 v1.2 and IS-05 v1.0, it'd be nice to support those.

Reply:
Tests now allow selection of IS-04 and (where appropriate) IS-05 versions.

Mock Node now only advertises `transporttype/` for versions v1.1+